feat: add docs for icons

AnnaRybkina opened the pull request:
Added initial docs for the icons. Based on https://www.fabric-ds.io/icons

Background: https://nmp-jira.atlassian.net/jira/software/projects/WARP/boards/15?selectedIssue=WARP-288

I have switch the order for Examples and Frameworks on this page as we have so many icons. But I'm glad to adjust. Take a look at the names also, do we want to keep them? In Fabric they were different but they didn't reflect the actual name that are used in the code.

Maybe as next iteration we can add theme switcher, maybe collapse examples. Better design? This version is pretty raw I would say. Do we want to link to the css docs from here so consumers can find icon colors easier?

![image](https://github.com/warp-ds/tech-docs/assets/37986637/fbf94403-8c34-439d-ae24-2fed28670312)

AnnaRybkina committed:
feat: add docs for icons

BalbinaK commented on docs/components/icons/elements.md:
````diff
@@ -0,0 +1,11 @@
+### Usage
+
+```html
+  <w-icon-bag16 class="s-icon"/>
````
Does this class always need to be set when using Warp icons? I'm asking because if that's the case, then maybe we should apply it inside each icon component 🤔

AnnaRybkina replied:
I just wanted to add it here to visualize that we have icon classes. Do not know if we should add it to the components. Icons will get current color otherwise, right?

BalbinaK replied:
Yes, icons will get current color from the wrapper. I'm unsure if the above example will be understood as "this is what you can do". At least when I saw it, I understood it as "this is what you should do when using icons". But maybe it's just me 😊  Could be worth mentioning somewhere here that icons default to `currentColor` and that colors can be changed using semantic color classes for icons listed here: https://warp-ds.github.io/css-docs/icon-color#icon-color.

AnnaRybkina replied:
That is what I asked here `Do we want to link to the css docs from here so consumers can find icon colors easier?` in the description. I'll add a link and remove the usage of the class

BalbinaK replied:
I was planning to answer questions from the PR description next ;) And yes, I think that it would be good to link to the css docs from this page.

BalbinaK commented on docs/components/icons/Example.vue:
```diff
@@ -0,0 +1,33 @@
+          <div class="text-center"><div class="mx-auto mb-8 s-bg rounded-4 h-56 flex items-center justify-center flex-col">
+            <component v-bind:is="icon" class="s-text"></component>
+          </div>
+          <div class="text-12">{{ name }}</div>
```
The name would always display as `IconNameSize` for each framework example in the docs, but that would only be correct for React icons. We could of course keep it like this until we handle selected framework state on each component page, but if you think it makes sense, we could also instead display just the icon name, like `Bag`, `Bookmark`, `AlertWarning` etc. Not consistent with the framework implementations still, but maybe a bit easier to read the list of icons. Just an idea :)

AnnaRybkina replied:
I raised this question in description. I don’t know which is best. Shall I adjust? I had a nice function for that that I can take back.

BalbinaK replied:
I personally would go for just the name, even if it's PascalCase for all framework examples. Just easier to look at icon names when browsing the list imo. Now that I think about it, it might be unnecessary to show the full name of each icon component in the table. I would feel it's enough to see how each component is used in the "Usage" section, meaning that it's always prefixed with `icon`, `w-icon` or `Icon` and postfixed with icon size. But maybe again, it's just me, and others would rather see the full name under each icon 🙂

AnnaRybkina replied:
https://github.com/warp-ds/tech-docs/pull/102/commits/b4b596f270abe797a5a304da4aff10b75f3c32b4

AnnaRybkina replied:
wdyt about the latest?

<img width="762" alt="image" src="https://github.com/warp-ds/tech-docs/assets/37986637/43efe6ca-2b0c-42e4-8c94-5a6f9bca7596">

BalbinaK replied:
I like that 😄 I think it's much easier to read.

BalbinaK commented on docs/components/icons/elements.md:
````diff
@@ -0,0 +1,11 @@
+## Import
+
+```js
+import '@warp-ds/icons/elements/icon-bag-16';
````
I thought icons were imported like this:
```
import '@warp-ds/icons/elements/bag-16'
```

If that's not correct, I will need to update the README in @warp-ds/icons 😅

AnnaRybkina replied:
I checked in the elements repo how we’ve done before. I can re-check

AnnaRybkina replied:
https://github.com/warp-ds/tech-docs/pull/102/commits/43743edb7715c6c5552fbac024c45f2b1bce8511

BalbinaK commented on docs/components/icons/vue.md:
````diff
@@ -0,0 +1,11 @@
+### Usage
+
+```html
+  <icon-bag-16 class="s-icon" />
````
Don't know if it makes any difference in Vue, but in the [README](https://github.com/warp-ds/icons#vue) in @warp-ds/icons it's documented that Vue icons should be used like `<icon-bag16>`. Would be good to align there, so just let me know if I should update the readme 😊

AnnaRybkina replied:
https://github.com/warp-ds/tech-docs/pull/102/commits/43743edb7715c6c5552fbac024c45f2b1bce8511

BalbinaK reviewed (commented):
Great stuff! Just a few questions :)

AnnaRybkina committed:
fix: adjust name of the icon

AnnaRybkina committed:
fix: adjustments after CR

AnnaRybkina commented:
@BalbinaK I've adressed the comments here https://github.com/warp-ds/tech-docs/pull/102/commits/43743edb7715c6c5552fbac024c45f2b1bce8511 in one commit. Please take a look 🙌

BalbinaK commented on docs/components/icons/Example.vue:
```diff
@@ -0,0 +1,35 @@
+      <div  class="grid gap-24 grid-cols-minmax-100px">
+        <div v-for="(icon, fullName) in mappedIconsBySize[size]" :key="fullName">
+          <div class="text-center"><div class="mx-auto mb-8 s-bg rounded-4 h-56 flex items-center justify-center flex-col">
+            <component v-bind:is="icon" class="s-text"></component>
```
> Maybe as next iteration we can add theme switcher

Adding a theme switcher on this page would be great as next step! Then I assume this `s-text` class would be replaced with `s-icon` and the ThemeSwitcher logic would handle it nicely for us?

AnnaRybkina replied:
I can change it now, good catch

BalbinaK commented on docs/components/icons/Example.vue:
```diff
@@ -0,0 +1,35 @@
+</script>
+
+<template>
+  <section v-for="(size) in Object.keys(mappedIconsBySize)" class="s-bg-primary-subtle rounded-8 p-24 mb-24">
```
It's a small thing but I wanted to mention it here as part of your question regarding theme switching. 
When we implement theme-switcher for icons, the `s-bg-primary-subtle` will be light green. If we used `s-bg-subtle` the background color difference won't be too big on theme change.
With s-bg-primary-subtle:
<img width="769" alt="Screenshot 2023-09-22 at 11 02 58" src="https://github.com/warp-ds/tech-docs/assets/41303231/333d6ff3-7224-4a14-8fe1-6494a0ed224c">
<img width="744" alt="Screenshot 2023-09-22 at 11 02 46" src="https://github.com/warp-ds/tech-docs/assets/41303231/48914a98-1cc2-489a-a0c5-25f85c8a88fc">

With s-bg-subtle
<img width="728" alt="Screenshot 2023-09-22 at 11 01 44" src="https://github.com/warp-ds/tech-docs/assets/41303231/d10fefab-5bb2-46f2-ad38-36bdd91b596e">
<img width="725" alt="Screenshot 2023-09-22 at 11 02 12" src="https://github.com/warp-ds/tech-docs/assets/41303231/a8c2775e-7e73-4ab0-8aa7-6cae2a39a097">

AnnaRybkina replied:
Hm, maybe it is better to use non-semantic color here? as I can't see the need of changing the color here, right? I haven't thought about the details when I added the color 🤔

BalbinaK replied:
I wouldn't mind it here as long as it's not too big of a change on theme switching, or at least that the class name is semantically correct (primary didn't seem like one 😄). I'd do whatever is easiest for you!

BalbinaK commented:
> Maybe as next iteration we can (...) maybe collapse examples. Better design?

I think it looks nice and that it's fine to have all examples listed like you show. We could wait for feedback from designers/users if it should be improved imo.

BalbinaK reviewed (approved):
I left small comments regarding the theme-switching but it's in no way a blocker as this page looks great! 🚀 🎸 🔥

AnnaRybkina committed:
fix: add themeswitcher

AnnaRybkina commented:
Based on your latest comments I added theme switcher and adjusted colors. Would you mind taking a look? https://github.com/warp-ds/tech-docs/pull/102/commits/aa33e53cb12330315bae90ce40b4a296f9373d70 It was not so many changes after all

AnnaRybkina committed:
fix: color of the icon name to work in dark mode

AnnaRybkina committed:
fix: add closing tag to the elements example
